### DISCUSSION TITLE
Closing Data

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#6137] [Archive]
Guys, I need to do the following: I need a column that has registered the closing date, ie when the user changes the status from open to closed, at that time I need to register the date in that column.

Any idea how this can be done?

Thanks.

**Comment by benhaynes** [comment]
Custom API Hook:

https://docs.directus.io/guides/api-hooks/#custom-api-hooks

**Comment by hugopvl** [comment]
Something like this?


module.exports = function registerHook({ exceptions }) {
        return {
                'items.update.before': async function (input, {collection, payload}) {
                        if (collection !== 'planing') return input;
						
						if (status == closed && close_date == ''){
						['close_date'] = date.now
						} else if (status != closed){
						['close_date'] = ''
						}
                        console.log(payload);
                        return input;
                }
        };
};

**Comment by hugopvl** [comment]
done and running:

if anyone needs it, stay here:

```js
module.exports = function registerHook({ exceptions, services }) {
        const { ItemsService } = services; return {
                'items.update.before': async function (input, { schema, collection, payload, item }) {
                        if (collection !== 'planeamento') return input;

                       const datanova = new Date; 
                       const estado = payload.status;

                        if (estado === 'draft') {
                                payload['data_fecho'] = datanova;
                        } else {
                                payload['data_fecho'] =null;
                        }
                        return input;
                },
        };
};
```
